### PR TITLE
Enhance dynamic buffer calculation and bug fixes

### PR DESCRIPTION
**Pull request by stephenxs**
**What I did**
Enhance dynamic buffer calculation and bug fixes

Signed-off-by: Stephen Sun <stephens@nvidia.com>
**Why I did it**

**How I verified it**
Run regression and vs test

**Details if related**
- Remove make_pair when calling emplace_back.
- The pool size isn't recalculated when a port with headroom-override but without speed or cable length configured being shutdown.
- Remove the current PG from the old referenced profile only if its name isn't empty, otherwise, an entry with an empty name will be inserted into m_bufferProfileLookup
- Don't try removing statically configured profiles with dynamic headroom from the APPL_DB because they were not programmed to APPL_DB.
- Setting a buffer PG to the same profile as it was causes the referenced dynamic profile unable to be removed once it isn't referenced anymore

**- Which release branch to backport (provide reason below if selected)**

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [x] 202012

### CODE REVIEW

**Comment by stephenxs** [comment]
retest this, please

**Comment by neethajohn** [comment]
retest this please

**Comment by stephenxs** [comment]
retest this please

**Comment by liat-grozovik** [comment]
retest this please

**Comment by stephenxs** [comment]
retest this please

File: cfgmgr/buffermgrdyn.cpp [hunk 1]
```diff
@@ -1418,7 +1424,7 @@ task_process_status BufferMgrDynamic::handleOneBufferPgEntry(const string &key,
                     }
                     else
                     {
-                        // In this case, we shouldn't set the dynamc calculated flat to true
+                        // In this case, we shouldn't set the dynamc calculated flag to true
```
Comment (neethajohn) [inline cfgmgr/buffermgrdyn.cpp:1427]: spell check 'dynamic'

**Review by neethajohn — APPROVED** [review]

File: cfgmgr/buffermgrdyn.cpp (hunk 1 above)
Comment (stephenxs) [reply cfgmgr/buffermgrdyn.cpp]: Fixed. Also fixed another one.

**Review by neethajohn — APPROVED** [review]

**Comment by stephenxs** [comment]
retest vsimage please

**Comment by stephenxs** [comment]
retest vs please

**Comment by neethajohn** [comment]
retest vs please

**Pull request merged**